Note Options Menu

theck13 opened the pull request:
### Fix
Move the ***History***, ***Share***, and ***Trash***/***Restore*** actions from the top app bar as well as the ***Pin***, ***Markdown***, ***Publish***, and ***Copy link*** actions from the ***Information*** bottom sheet to the overflow menu.  Also, move ***Search*** to  the first/leftmost action in the app bar.  Note the ***Copy link*** action will only be enabled when the ***Publish*** action is enabled and checked.  These changes do no include updates to the snackbar color and text.  Those will be included in a subsequent pull request.  See the screenshots below for illustration.

![note_options_menu_phone](https://user-images.githubusercontent.com/3827611/74001733-f4563a80-4929-11ea-828d-9b9dadcbf7d8.png)

![note_options_menu_tablet](https://user-images.githubusercontent.com/3827611/74001737-f6b89480-4929-11ea-90c9-e649cdddd2a9.png)

### Test
Since the actions are different on phones and tablets, it's best to test both device form factors to verify the correct actions are enabled/disabled and visible/invisible as shown in the screenshots above.
1. Tap any note in list.
2. Notice ***History***, ***Share***, and ***Trash*** actions in overflow menu.
3. Notice ***Pin***, ***Markdown***, ***Publish***, and ***Copy link*** actions in overflow menu.
4. Tap ***Pin*** action in overflow menu.
5. Notice note is pinned to top of list.
7. Tap ***Markdown*** action in overflow menu.
8. Notice markdown is enabled for note.
9. Tap overflow menu action.
10. Notice ***Pin***, ***Markdown***, ***Share***, ***History***, and ***Publish*** actions enabled.
11. Notice ***Trash*** action enabled.
12. Show markdown view of note.
    a. Tap ***Preview*** tab on phone.
    b. Tap ***Show Markdown*** action on tablet.
13. Tap overflow menu action.
14. Notice ***Pin***, ***Markdown***, ***Share***, ***History***, ***Publish***, and ***Copy link*** actions disabled.
15. Notice ***Trash*** action enabled.
16. Tap ***Publish*** action in overflow menu.
17. Notice ***Copy link*** action enabled after publish succeeds.
18. Open navigation drawer.
19. Tap ***Trash*** item in navigation drawer.
20. Tap any note in list.
21. Tap overflow menu action.
22. Notice ***Pin***, ***Markdown***, ***Share***, ***History***, ***Publish***, and ***Copy link*** actions disabled.
23. Notice ***Restore*** action enabled.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in 06fa4ad with:
> Moved note options from the action bar to the ellipsis menu

SylvesterWilmott commented:
Nice work @theck13 ! It looks really good.

A few minor things:
- The more icons' colour doesn't match the other icons in the appbar
- The menu background colour (dark theme) should use one of our dark colours. Based on it's 8dp elevation it should be `#323436`
- How are the checkboxes aligned? Are we able to right-align them 16dp from the menu edge?

Regarding the trash, I like what you've done for the consistency of the options between trashed notes and non-trashed notes.

Tablet also looks good!

theck13 commented:
I updated the toolbar icon colors in 3c68bdd0.  See the screenshots below for illustration.

![note_options_menu_phone_icon](https://user-images.githubusercontent.com/3827611/74050729-d7565180-4993-11ea-9303-d1fc39acbf2d.png)

I updated the background color in 40ab64b3.  Unfortunately, that overlays the group dividers.  Is that alright?  See the screenshots below for illustration.

![note_options_menu_phone_popup](https://user-images.githubusercontent.com/3827611/74050545-86465d80-4993-11ea-9dd6-ec9c3ae1f804.png)

SylvesterWilmott commented:
Thanks, @theck13 

Let's still attempt to align checkboxes in another PR
